src/mats_l1_processing/read_parquet_functions.py

```diff
@@ -211,14 +213,20 @@ def read_ccd_data_in_interval(
     if stop.tzinfo is None:
         stop.replace(tzinfo=timezone.utc)
 
+    filterlist = (
+        (ds.field("EXPDate") >= Timestamp(start))
+        & (ds.field("EXPDate") <= Timestamp(stop))
+    )
+    if filter != None:
+        for variable in filter.keys():
+            filterlist = filterlist & ((ds.field(variable) >= filter[variable][0]) &
+                                       (ds.field(variable) <= filter[variable][1]))
+
     table = ds.dataset(
         path,
         filesystem=filesystem,
-    ).to_table(filter=(
-        (ds.field("EXPDate") >= Timestamp(start))
-        & (ds.field("EXPDate") <= Timestamp(stop))
-    ))
-    dataframe = table.to_pandas().reset_index()
+    ).to_table(filter=filterlist)
+    dataframe = table.to_pandas().reset_index().set_index('TMHeaderTime')
```

Comment:
I don't think you need to do both `reset_index()` and `set_index(...)`. The latter should be enough, if you want a time-index on what is returned.
```suggestion
    dataframe = table.to_pandas().set_index('TMHeaderTime')
```

Reply:
no you need both otherwise EXPDate  disappears for some reason

Reply:
I think it would be a good idea to mention that this function sets the index, and to what, in the doc-string.

Reply:
A good place for it is where the return type is described, _i.e._
```python
DataFrame:      The CCD data, indexed by `TMHeaderTime`
```

Also consider applying `.sort_index()` after `.set_index(...)`.